### DISCUSSION TITLE
zh/guide/drivers/common

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#17] [Announcements]
# zh/guide/drivers/common

AList v3的文档

https://alist.nn.ci/zh/guide/drivers/common.html

**Comment by xx-zhang** [comment]
怎么使用webdav或者alist携带的上传功能呢；

**Reply by 3555555678** [reply]
怎么打包下载 有详细教程吗

**Reply by changchichung** [reply]
好奇怎麼上傳 +1
在 alist 頁面找不到有上傳的選項丫？

**Reply by KDreamelf** [reply]
右下角有更多的按钮，点击它，上方弹出的按钮包含上传。

**Reply by mlkt** [reply]
操作功能藏在右下角这设计真是绝了，好像是怕别人看到似的，这难道不应该显示在顶端作为工具栏吗

**Reply by zhangjiuliang** [reply]
那你自己改呀，不是开源的吗，改完了你发出来

**Comment by 3555555678** [comment]
怎么打包下载 有详细教程吗

**Comment by yubanmeiqin9048** [comment]
使用非标准端口的的alist地址可以使用cloudflare workers吗？

**Reply by xhofe** [reply]
不可以

**Comment by uniStark** [comment]
如果没有公网nas，推荐使用服务器安装alist做映射使用最佳

**Comment by ponponon** [comment]
所以怎么挂载？怎么实现『唯一标识，即要挂载到的位置，如果要挂载到根目录，就是 /』？

**Reply by KDreamelf** [reply]
你可以理解为，虚拟文件夹。例如挂载到dreamelf/oneDrive，那么你的所有文件会挂载到dreamelf/oneDrive/目录下。目录名称自然是不能重复的

**Comment by cnliucheng** [comment]
请问如何挂载SMB

**Comment by LevineWoo** [comment]
win版怎么设置代理？比如127.0.0.1:7890

**Reply by KDreamelf** [reply]
你是指下载代理？下载是由客户端指定的。
如果是指服务器到网盘的代理，则需要为服务器配置代理服务器，然后确认alist的服务，然后对该服务进行代理。亦或者，使用docker安装，然后将docker进行代理

**Comment by GaryRandom** [comment]
有个关于下载的问题，我在nas上搭建了alist，在笔记本上管理alist，下载我想把alist挂载的文件下载在NAS里，但alist的各种代理方式都是下载到笔记本上。请问这个大家怎么解决的？

**Comment by sunjanesy** [comment]
Cloudflare Workers不是被FW将近快1年了,国内根本用不上的,建议中文教程这个可以删掉了

**Reply by KDreamelf** [reply]
我就说，我弄好workers之后，不开梯子下载速度几乎莫得，而且即便强行开梯子下载，worker的请求量也爆炸。如此折腾还不如上oneDrive等国外网盘的官网下载（以上都是跑IDM的情况哈）

**Reply by sunjanesy** [reply]
大概中文教程不单指面向内地,但是最起码一标注下对吧,墙内无效
又或者作者们都是身处国外,也没几个真用到代理,来反馈问题

**Reply by Winyer** [reply]
workers在国内某些区域还是能用的，只是速度不如在国外那么理想

**Comment by 57334920** [comment]
请问是否支持挂载腾讯云盘？

**Reply by 57334920** [reply]
新版本已支持。

**Comment by Jerry12228** [comment]
用cloudflare workers代理时下载文件需要挂梯子???
https://alist-proxy.jerrybest.workers.dev/
测试文件:
https://pan.jerry.ink/proxy-test/line-13-0-1.apk

**Reply by Jerry12228** [reply]
加自定义域就行了QAQ

**Comment by 77931214** [comment]
请问，配置了反向代理（默认的5244端口），还可以使用 Cloudflare Workers 做代理下载吗？

**Reply by Jerry12228** [reply]
可以啊，ADDRESS填反代后的域名即可

**Reply by 77931214** [reply]
我看说明，只支持默认端口呀，80/443？

**Comment by tbx8564a** [comment]
请问怎么可以把不同网盘的文件挂载到同一个挂载路径？设置相同路径后显示：
Failed update storage in database: UNIQUE constraint failed: x_storages.mount_path

**Reply by tangxiadi** [reply]
同问

**Reply by dh53128** [reply]
我是定义为/1 /2 ，然后别名里面把/1和/2 挂到一个目录。不知道有没有更好的办法

**Comment by sunlyc** [comment]
希望能增加联通云盘， 联通云盘出了一个联通云硬盘，可以挂载到本地， 如果alist能集成进去就太好了。

**Comment by coco-coc** [comment]
希望能添加一个Linkbox网盘它内存够多🙏

**Comment by fgfg163** [comment]
pikpak速度有点慢，于是我就想用Cloudflare Workers加速一下pikpak，中间踩坑很多
测试了半天Cloudflare Workers，斥资2个域名和1个腾讯轻量云主机，最终得到如下结论：

测试条件：
阿里云购买的2个域名，1个xxxx.cn在阿里云解析，一个xxxx.com托管到cloudflare解析
腾讯轻量云北京机房，公网ipv4没有ipv6
联通5g移动网络+5g cpe，支持ipv6，默认运营商的dns
openwrt主路由，在这里修改网络配置
家用笔记本电脑
小米手机使用5g网，使用dns changer和chrome浏览器测试

腾讯云http所有端口都会被拦截跳转到备案页，https的443端口直接停止响应，其他端口可以用
Cloudflare 实际上有若干https端口可以用，自行参考文档。我这里测试Cloudflare Workers访问到https的8443端口可以通畅，就是nginx代理到8443端口并且nginx里配置好ssl证书。
在阿里云配置好域名解析，域名使用https://xxxx.cn:8443;

cloudflare配置好Cloudflare Workers和域名和路由，域名是xxxx.com，在开梯的情况下可以正常访问。
Cloudflare Workers自定义域名我写的https://worker.xxxx.com
Cloudflare Workers脚本里的ADDRESS我写的https://xxxx.cn:8443，对应上面腾讯轻量云开的https端口

然后我关掉梯子反复修改openwrt路由，拔插网线（清除电脑的dns缓存）得到一些结论：
在不用梯的情况下
1. 运营商dns无法解析Cloudflare Workers送的域名和自定义域名xxxx.com，其他托管的子域名解析断断续续不稳定
2. Cloudflare Workers送的域名可以解析但是无法访问
3. Cloudflare Workers自定义域名，在使用谷歌dns后可以正常使用了
4. 禁止解析ipv6，整体速度更快了

总结：
1. 要是你有自定义dns或者openwrt里配了自定义dns，Cloudflare Workers可以用。限制的都是dns解析
2. 国内服务器可以试试8443端口

**Reply by fgfg163** [reply]
再次测试，我在Cloudflare Workers上配置的步骤
1. 托管xxxx.com域名到Cloudflare
2. 添加A解析pan.xxxx.com指向腾讯轻量云服务器，关闭
3. 添加CNAME解析panproxy.xxxx.com指向pan.xxxx.com并开启代理，提供给Cloudflare Workers使用
4. 创建Cloudflare Workers，ADDRESS填写https://panproxy.xxxx.com:8443
5. 添加自定义域panworker.xxxx.com，添加自定义路由panworker.xxxx.com/*
6. 在本地nslookup测试下解析，测试下worker访问是否正常，不正常就不用继续了
6. 腾讯云部署好服务，开放8443端口并且添加证书，各种配置

测试时打开10多次视频pikpak就提示我访问次数过多了，需要注意

**Comment by xsd116** [comment]
能支持腾讯微云吗

**Comment by qinxianyuzou** [comment]
请问有手动刷新目录的功能吗？因为缓存时间设置得比较长，所以要等很久才会同步，影响效率

**Comment by lovesnsfi** [comment]
**不好意思，我是新手**
我想问一下，那个 **签名** 的功能 干什么用的啊？

**Comment by ZC-TigerRoot** [comment]
在使用CF-workers作代理后，不加速的works自带链接可以正常下载，但是自定义的加速链接报错1101
测试链接
https://alist-proxy.tigerroot.workers.dev/onedrivecn/%E8%A3%85%E9%A5%B0/logo.jpg?sign=DPwtSEa87URv_J2OBl3cZFc85c8b0EKy-Z3zfx6Bbws=:0
https://alist.tigerroot.cn/onedrivecn/%E8%A3%85%E9%A5%B0/logo.jpg?sign=DPwtSEa87URv_J2OBl3cZFc85c8b0EKy-Z3zfx6Bbws=:0
以下是CF-works抓出来的日志
```json
{
  "outcome": "exception",
  "scriptName": "alist-proxy",
  "diagnosticsChannelEvents": [],
  "exceptions": [
    {
      "name": "SyntaxError",
      "message": "Unexpected token 'e', \"error code: 1016\" is not valid JSON",
      "timestamp": 1707835944361
    }
  ],
  "logs": [],
  "eventTimestamp": 1707835944345,
  "event": {
    "request": {
      "url": "https://alist.tigerroot.cn/onedrivecn/%E8%A3%85%E9%A5%B0/Wallpaper/%E7%BB%BF%E8%8C%B5%E5%B9%BD%E8%8D%89%E3%80%90%E5%8E%9F%E7%89%88%E3%80%914K%E5%8A%A8%E6%80%81%E5%A3%81%E7%BA%B8_3049.mp4?sign=REDACTED=:0",
      "method": "GET",
      "headers": {
        "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7",
        "accept-encoding": "gzip",
        "accept-language": "zh-CN,zh;q=0.9",
        "cf-connecting-ip": "117.154.92.131",
        "cf-ray": "854de51c190a22a8",
        "cf-visitor": "{\"scheme\":\"https\"}",
        "connection": "Keep-Alive",
        "host": "alist.tigerroot.cn",
        "sec-ch-ua": "\"Not A(Brand\";v=\"99\", \"Google Chrome\";v=\"121\", \"Chromium\";v=\"121\"",
        "sec-ch-ua-mobile": "?0",
        "sec-ch-ua-platform": "\"Windows\"",
        "sec-fetch-dest": "document",
        "sec-fetch-mode": "navigate",
        "sec-fetch-site": "cross-site",
        "sec-fetch-user": "?1",
        "upgrade-insecure-requests": "1",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36",
        "x-forwarded-proto": "https",
        "x-real-ip": "117.154.92.131"
      },
      "cf": {
        "clientTcpRtt": 74,
        "longitude": "114.26810",
        "latitude": "30.58900",
        "tlsCipher": "AEAD-AES128-GCM-SHA256",
        "continent": "AS",
        "asn": 9808,
        "clientAcceptEncoding": "gzip, deflate, br",
        "country": "CN",
        "tlsClientAuth": {
          "certIssuerDNLegacy": "",
          "certIssuerSKI": "",
          "certSubjectDNRFC2253": "",
          "certSubjectDNLegacy": "",
          "certFingerprintSHA256": "",
          "certNotBefore": "",
          "certSKI": "",
          "certSerial": "",
          "certIssuerDN": "",
          "certVerified": "NONE",
          "certNotAfter": "",
          "certSubjectDN": "",
          "certPresented": "0",
          "certRevoked": "0",
          "certIssuerSerial": "",
          "certIssuerDNRFC2253": "",
          "certFingerprintSHA1": ""
        },
        "tlsClientHelloLength": "736",
        "tlsExportedAuthenticator": {
          "clientFinished": "b43a060899200f2c35b459e069602ba55e7ff7ac5c3fb8639bebfe0b621d56ca",
          "clientHandshake": "306ee4d02666b4cc22476b42d64d72671e1e7e18d1bf146ff0bbbdd21d8b7f1e",
          "serverHandshake": "6da094ed8da4f5fe81a8d614be14a1472d69fa088648c8959ae5cb2067e88c0d",
          "serverFinished": "925e2d8cc10a486362085371c182069a0c16aaa1eada249d6bdf7eee484fd525"
        },
        "tlsVersion": "TLSv1.3",
        "colo": "HKG",
        "timezone": "Asia/Shanghai",
        "verifiedBotCategory": "",
        "edgeRequestKeepAliveStatus": 1,
        "requestPriority": "weight=256;exclusive=1",
        "tlsClientRandom": "7NlsVf3i8ysdUhnWLLIjWOeQR+eVTjhU/BK+23PFGWk=",
        "city": "Wuhan",
        "region": "Hubei",
        "regionCode": "HB",
        "asOrganization": "China Mobile",
        "httpProtocol": "HTTP/2"
      }
    },
    "response": {
      "status": 500
    }
  },
  "id": 0
}
```

**Comment by hubgha-hub** [comment]
在openwrt x86上面把alist从3.29替换到3.32后，添加的本地存储文件夹打开一直转圈，而添加的smb存储能正常访问，请问这是怎么回事？谢谢！

**Comment by xianger2017** [comment]
这个有签名成功的吗

**Comment by chenweiliang6** [comment]
302重定向共享使用，有封禁账户的风险，共享使用具体是指什么，如果是多个不同的设备alist挂同一个天翼云盘，这算共享吗

**Comment by at86397388** [comment]
建议增加苹果的  icloud drive 驱动
